#23 library button

Secur1ty-21 opened the pull request.

svetSanna committed:
Добавлен файл цветов

svetSanna committed:
1. Добавлена ActiveButton из patch
2. Добавлена функция root в MainActivity.kt
3. Исправлен class на object в VolleyColor

svetSanna committed:
4. Добавлен шрифт Actay Wide Bold

svetSanna committed:
4. кнопка GradientButton

svetSanna committed:
6. Добавлены шрифты Hero

svetSanna committed:
7. Добавлены иконки
8. Сделан вариант градиентной кнопки с указанием позиции (top, right) и Id картинки

svetSanna committed:
9. outlinedgradientbutton
10. checkedGradientButton
11. CheckedGradientButtonTopImage - пока только IsChecked = true

svetSanna committed:
12. Добавлены изображения для кнопок
13. ButtonConfirmLevel, ButtonLevelDown, ButtonLevelUp, CheckedGradientButtonTopImage, CheckedGradientButtonRightImag, CheckGradientButton, OutlinedGradientButton, ActiveGradientButton, OutlinedActiveButton, ActiveButton

svetSanna committed:
14. Убрала сохранение сотcояний IsChecked => все функции сделала с аннотацией @Stable
15. сделала во всех функциях onClick последним параметром

svetSanna committed:
16. ButtonsGroup, GroupButtonsForLevel

svetSanna committed:
размеры

svetSanna committed:
группы

svetSanna committed:
сделала основные кнопки и группы. надо протестировать логику и сделать Preview отдельно для каждой кнопки, группы

svetSanna committed:
добавила кнопки ActiveButonSmallText("Add payment") и ActiveButtonMap("Map")

svetSanna committed:
картинки заменила на png

svetSanna committed:
переложила в другие папки

svetSanna committed:
Merge branch 'develop' into 23-library-button

svetSanna committed:
шрифты откорректировала

svetSanna committed:
картинки на кнопках исправила

svetSanna committed:
кнопку исправила

svetSanna committed:
preview

svetSanna committed:
preview сделала для всех элементов и групп отдельно

svetSanna committed:
add payment

Commit:
#23 Fix detekt rules for review visibility

Commit:
Merge branch '23-library-button' of github.com:o1isrta/VolleyBolleyAndroid into 23-library-button

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt

Commit:
Merge remote-tracking branch 'origin/develop' into 23-library-button

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/presentation/ui/model/VolleyColor.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/model/VolleyText.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/model/VolleyTypography.kt

Commit:
#23 Improve preview visibility

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+    fun OutlinedActiveButtonSmallText(
+        modifier: Modifier = Modifier,
+        text: String,
+        paddingValues: PaddingValues = PaddingValues(16.dp, 8.dp, 16.dp, 8.dp),
```
(Не крит) Думаю стоит убрать возможность у этой кнопки менять paddingValues, т.к. только этим отличается от OutlinedGradientButton

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+    fun CheckGradientButton(
+        modifier: Modifier = Modifier,
+        text: String,
+        isChecked: Boolean = false,
```
(Не крит) По умолчанию в isChecked лучше не выставлять, т.к. суть этой кнопки как раз в переключении и поэтому этот параметр желательно требовать

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+     */
+    @Composable
+    @Stable
+    fun CheckedGradientButtonRightImage(
+        modifier: Modifier = Modifier,
+        text: String,
+        isChecked: Boolean = false,
+        paddingValues: PaddingValues = PaddingValues(start = 16.dp, top = 12.dp, end = 16.dp, bottom = 12.dp),
+        onClick: () -> Unit
+    ) {
```
(Не крит) Думаю можно переиспользовать ActiveGradientButton и OutlinedGradientButton, добавив в них параметр картинки

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+    }
+
+    @Stable
+    interface ButtonItem {
```
(Крит) Лучше сделать data классом и дополнительно к Stable пометить как Immutable

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+        isExclusive: Boolean = true,
+        onSelected: (Int) -> Unit
+    ) {
+        var checkSet = emptySet<Int>()
+        items.forEach { item ->
+            if (item.isChecked) {
+                checkSet = checkSet + item.position
+            }
+        }
+        val state = rememberSaveable { mutableStateOf(checkSet) }
```
(Крит) Тут лучше передать, какие кнопки отмечены, а какие нет, на внешнее управление. В параметрах функции уже всё есть для того, чтобы тот, кто вызывает функцию, мог определить, какая кнопка отмечена, а какая нет

svetSanna replied:
Ой, что-то я не поняла. То есть checkSet и state вообще здесь не нужны? А где будет обрабатываться логика нажатия (в случае IsExclusive = true или =false)?

Secur1ty-21 replied:
Да, лучше передать эту логику пользователю библиотеки. Логика isExclusive тоже уйдет на сторону пользователя

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -0,0 +1,1298 @@
+     */
+    @Composable
+    fun GroupButtonsForTourneyType(checkId: Int = 1, modifier: Modifier, onSelected: (Int) -> Unit) {
+        var selectedButton by remember { mutableIntStateOf(checkId) }
```
(Крит) Лучше убрать, т.к. нельзя будет поменять выбранную кнопку через внешнее управление после инициализации

svetSanna replied:
Не поняла, что убрать?

Secur1ty-21 replied:
Переменную с remember
```kotlin
var selectedButton by remember { mutableIntStateOf(checkId) }
```

Secur1ty-21 reviewed (commented):
Отличная работа! То что отмечено "Крит" необходимо поправить, то что отмечено "не крит" просто рекомендации и исправлять не обязательно

svetSanna committed:
константы текстовые

svetSanna committed:
в OutlinedActiveButtonSmallText() убрана возможность менять paddingValues

svetSanna committed:
в CheckGradientButton() убрано значение по умолчанию для isChecked

svetSanna committed:
ButtomItem переделан в data class

svetSanna committed:
убрала isExclusive, CheckSet и state (то есть логику) из ButtonsGroup()

svetSanna committed:
1 убрала isExclusive, CheckSet и state (то есть логику) из GroupButtonsForTourneyType
2 добавила кнопки-слайдеры
3 попробовала ActiveGradeintButton сделать не через box, а через button. Надо отпределиться, как делать

svetSanna committed:
сделала ActiveGradientButton, CheckedGradientButtonRightImage,CheckedGradientButtonTopImage через Button (с эффектом нажатия и без уголков при мерцании)

nemislimus committed:
Merge branch 'develop' into 23-library-button

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/presentation/MainActivity.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/VolleyContainersRootTransparent.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/VolleyTextFieldAttribute.kt

nemislimus committed:
#23 Свёл конфликты и адаптация кода.

nemislimus committed:
#23 fixes-according-detekt-1

nemislimus committed:
#23 fixes-according-detekt-2

nemislimus reviewed (approved):
Я веточку когда-то давно уже смотрел, давал рекомендации.
И немного по конфликтам помогал сводить.
